### PR TITLE
Update cit filtering to use append instead of replace method

### PR DESCRIPTION
**Pull request by jjerger**
Allows for easier pipeline definitions in terms of defining additional tests

### CODE REVIEW

File: concourse/pipelines/linux-image-build.jsonnet [hunk 1]
```diff
@@ -235,7 +235,11 @@ local imgpublishjob = {
   trigger:: if tl.env == 'testing' then true
   else false,
 
-  citfilter:: '^(cvm|livemigrate|suspendresume|loadbalancer|guestagent|hostnamevalidation|imageboot|licensevalidation|network|security|hotattach|lssd|disk|packagevalidation|ssh|metadata|vmspec)$',
+  // Use citfilter list as default; append citsuites if not nil
+  // citsuites must be in the valid regex concatenation format of 'item1|item2|item3'
+  citsuites:: '',
```
Comment (a-crate) [inline concourse/pipelines/linux-image-build.jsonnet:240]: additionalcitsuites would be a more descriptive name

File: concourse/pipelines/linux-image-build.jsonnet [hunk 2]
```diff
@@ -235,7 +235,11 @@ local imgpublishjob = {
...
+  citfilter:: if tl.citsuites == '' then '^(cvm|livemigrate|suspendresume|loadbalancer|guestagent|hostnamevalidation|imageboot|licensevalidation|network|security|hotattach|lssd|disk|packagevalidation|ssh|metadata|vmspec)$'
+  else '^(cvm|livemigrate|suspendresume|loadbalancer|guestagent|hostnamevalidation|imageboot|licensevalidation|network|security|hotattach|lssd|disk|packagevalidation|ssh|metadata|vmspec|%s)$' % tl.citsuites,
```
Comment (a-crate) [inline concourse/pipelines/linux-image-build.jsonnet:242]: `citfilter:: '^(cvm|livemigrate|suspendresume|loadbalancer|guestagent|hostnamevalidation|imageboot|licensevalidation|network|security|hotattach|lssd|disk|packagevalidation|ssh|metadata|vmspec%s)$' % '|'+tl.citsuites` would be equivalent and less repetitive

File: concourse/pipelines/linux-image-build.jsonnet (hunk 1 above)
Comment (jjerger) [reply concourse/pipelines/linux-image-build.jsonnet]: Done

**Review by a-crate — APPROVED** [review]

**Pull request merged**